tests/test_unit_db_models.py

```diff
@@ -328,3 +330,65 @@ async def test_insert_task_with_meta_none(
         wf = Workflow(name="my wfl", project_id=project.id)
         args = dict(arg="test arg")
         await wf.insert_task(t0.id, db=db, args=args)
+
+
+async def test_delete_dataset_cascade(db, MockCurrentUser, project_factory):
```

Comment:
Remove this test, after moving whatever is needed to one of the `test_dataset_api.py` tests